storage/HashTable.hpp

```diff
@@ -127,14 +116,18 @@ class HashTable : public HashTableBase<resizable,
   static constexpr bool template_force_key_copy = force_key_copy;
   static constexpr bool template_allow_duplicate_keys = allow_duplicate_keys;
 
-  // A surrogate hash value for empty buckets. Keys which actually hash to this
-  // value should have their hashes mutated (e.g. by adding 1). We use zero,
-  // since we will often be using memory which is already zeroed-out and this
-  // saves us the trouble of a memset. This has some downside, as the std::hash
-  // function we use is the identity hash for integers, and the integer 0 is
-  // common in many data sets and must be adjusted (and will then spuriously
-  // collide with 1). Nevertheless, this expense is outweighed by no longer
-  // having to memset large regions of memory when initializing a HashTable.
+  // Some HashTable implementations (notably LinearOpenAddressingHashTable)
```

Comment:
Since this is a key "gateway" file for anyone looking at hashtables for operators, could we add a more detailed comment field here describing briefly the hash tables, and specifically the cases when the SimpleScalarSeparateChainingHashTable is used. Thanks!